src/Networking/NexusMods.Networking.NexusWebApi/RunUpdateCheck.cs

```diff
@@ -49,28 +49,70 @@ public static async Task<PerFeedCacheUpdaterResult<PageMetadataMixin>> CheckForM
     {
         // Note(sewer): Undetermined items may be removed items from the site; or
-        // caused by programmer error, so wr should log these whenever possible,
+        // caused by programmer error, so we should log these whenever possible,
         // but they should not cause a critical error; in case it's simply the result
         // of mod removal such as DMCA takedown.
-        foreach (var mixin in result.UndeterminedItems)
+        
+        // Note(sewer): The semaphore below limits the maximum number of requests to the Nexus API
+        // (UpdateModPage) that can be in transit at any time. i.e., We will process max
+        // 'semaphoreCount' requests at a time. The number specified in the constructor
+        // is the maximum number of requests that can be in transit at any given time.
+        // It is completely arbitrary and can be safely changed. If the API ever adds 
+        // rate limits, this may need tweaking.
+        //
+        // When this code is important is if the user doesn't use the App for a long time,
+        // i.e. past the cache expiry time. Currently, that's 28 days. Past that point,
+        // when checking for an update, all the mod pages will need refreshing; 
+        // which will kick in this rate limiter. Otherwise, outside of that, concurrent
+        // requests here are very unlikely.
+        using var semaphore = new SemaphoreSlim(16);
+        var tasks = new List<Task>();
+
+        // Helper function to process a single mixin with error handling
+        async Task ProcessMixin(SemaphoreSlim sema, PageMetadataMixin mixin, bool isUndetermined)
         {
             try
             {
-                await UpdateModPage(db, tx, gqlClient, cancellationToken, mixin);
+                var isTaken = await sema.WaitAsync(Timeout.Infinite, cancellationToken);
+                try
+                {
+                    await UpdateModPage(db, tx, gqlClient, cancellationToken, mixin);
+                }
+                finally
+                {
+                    if (isTaken)
+                        sema.Release();
+                }
             }
             catch (Exception e)
             {
-                var id = mixin.GetModPageId();
-                logger.LogError(e, "Failed to update metadata for Mod (GameID: {Page}, ModId: {ModId})", id.GameId, id.ModId);
+                if (isUndetermined)
+                {
+                    var id = mixin.GetModPageId();
+                    logger.LogError(e, "Failed to update metadata for Mod (GameID: {Page}, ModId: {ModId})", id.GameId, id.ModId);
+                }
+                else
+                {
+                    // Rethrow for non-undetermined items as these should be truly exceptional
+                    throw;
+                }
```

Comment:
I don't think you need two try catch here. I would declare `isTaken = false;` outside then do a single try catch and finally with the  `if (isTaken) sema.Release();`

Reply:
All done 👏